interview-questions.md

```diff
@@ -5,40 +5,64 @@ Answer the following questions. First, without external resources. Challenge you
 1. Why would you use the method super()?
 
   Your answer:
+      Super is used in conjunction with "constructor" which will call functions from a parent class.
 
   Researched answer:
+      When used in a constructor, the super keyword appears alone and must be used before the "this" keyword is used. The super keyword can also be used to call functions on a parent object. super() will call the constructor of its parent class. This is required when you need to access some variables from the parent class. In React, when you call super with props. React will make props available across the component through this.props .
+        - taken from LearnAcademy lesson plan + MDN
 
 
 
 2. What is a virtual DOM?
-
   Your answer:
+      I think it stands for domain object manager and its purpose is to manage how the code translates to the interface.
 
   Researched answer:
+      In React, for every DOM object, there is a corresponding “virtual DOM object.” A virtual DOM object is a representation of a DOM object, like a lightweight copy.
+
+      A virtual DOM object has the same properties as a real DOM object, but it lacks the real thing’s power to directly change what’s on the screen.
+
+      Manipulating the DOM is slow. Manipulating the virtual DOM is much faster, because nothing gets drawn onscreen. Think of manipulating the virtual DOM as editing a blueprint, as opposed to moving rooms in an actual house.
+        - taken from CodeCademy!
 
 
 
 3. When creating a basic (stateless) class component in React, what are the necessary elements needed to display "Hello World" in the browser?
 
   Your answer:
+        class Name extends Component {
+          render (){
+            <h1> Hello World</h1>
+          }
+        }
```

Comment:
Nice! I appreciate you showing the code we would use. 

I think in an interview setting, we might not have the ability to type all of this out (Especially if its a phone interview), I would try and put it into words to describe the steps. E.g. "I would need a class, with a name, that extends Component. I would need a render and a return" etc. Additionally, I would want to also maybe consider if we would need to include the 'imports' and the 'export'. 

Again, great answer!